linux-kernel: update to 6.10.14

KexyBiscuit opened the pull request:
Topic Description
-----------------

- linux-kernel: unset `CONFIG_USB_ONBOARD_DEV` to fix boot livekit on MacBook Pro Mid 2010
- kernel-tools: update to 6.10.14
    - Synchronize patches from linux-kernel.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: update to 6.10.14
    - Nearly identical with 6.10.14-rc1.
    - Track patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.10.14, current HEAD is c588e0d40f688b9df73515b354a111b81240977e.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: update Power ISA config for 6.10.14-rc1
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: update config for 6.10.14-rc1
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- kernel-tools: update to 6.10.14-rc1
    - Synchronize patches from linux-kernel.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux+kernel: update to 6.10.14
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: update to 6.10.14-rc1
    - Routine patch due diligence for upstream releases.
    - Now following contribution guidelines of Android common kernels <https://android.googlesource.com/kernel/common/+/refs/heads/android-mainline/README.md>.
    - Update patch from kernel_compiler_patch.
    - Add a fix for PixArt device detection.
    - Track patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.10.14-rc1, current HEAD is 4f89320f66da12f2f34181928cbf3a5e79ad5996.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

Package(s) Affected
-------------------

- kernel-tools: 6.10.14-1
- linux+kernel: 3:6.10.14
- linux-kernel-${__VER}: 1:6.10.14-1

Security Update?
----------------

No

Build Order
-----------

```
#buildit linux-kernel linux+kernel kernel-tools
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

KexyBiscuit committed:
linux-kernel: update to 6.10.14-rc1

- Routine patch due diligence for upstream releases.
  - Now following contribution guidelines of Android common kernels <https://android.googlesource.com/kernel/common/+/refs/heads/android-mainline/README.md>.
  - Update patch from kernel_compiler_patch.
  - Add a fix for PixArt device detection.
  - Track patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.10.14-rc1, current HEAD is 4f89320f66da12f2f34181928cbf3a5e79ad5996.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux+kernel: update to 6.10.14

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: update to 6.10.14-rc1

- Synchronize patches from linux-kernel.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux-kernel: update config for 6.10.14-rc1

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux-kernel: update Power ISA config for 6.10.14-rc1

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit commented:
Smoke testing passed on AMD64 generation 2, AMD64 generation 1 and Armv8-A generation 2 Hyper-V VMs.

![image](https://github.com/user-attachments/assets/7db5ee61-cf60-4794-b6a4-ae4411ec9209)
![image](https://github.com/user-attachments/assets/dc8b9091-de97-4854-838b-100ca3b6618b)
![image](https://github.com/user-attachments/assets/fee8575f-eba8-40f2-9076-692039d2ad01)

KexyBiscuit commented:
Smoke testing passed on AMD64 VMware VM.

- Product: VMware® Workstation 17 Pro
- Version: 17.6.0 build-24238078
- Host OS version: Windows 11 Enterprise, 64-bit (Build 27718.1000) 10.0.27718
- Accelerate 3D graphics: on
- Firmware type: UEFI
- Host VBS Mode: on

![image](https://github.com/user-attachments/assets/a6c05c4a-e65e-4325-803c-331ce23a12f0)

KexyBiscuit commented:
Smoke testing passed on 战神Z8-KP7S2.

![图片](https://github.com/user-attachments/assets/0c0c3f34-5faf-428e-80fe-6b0425db5045)

KexyBiscuit commented:
Smoke testing passed on RedmiBook 16 (锐龙版).

![图片](https://github.com/user-attachments/assets/efc397a4-d155-43d3-9752-6380612c02d2)

KexyBiscuit commented:
Smoke testing passed on Surface Pro (5th Gen).

![图片](https://github.com/user-attachments/assets/4ad1b77e-ceb1-4f9d-9d5d-2ec65e5c5cc5)

KexyBiscuit commented:
Smoke testing passed on ROG Ally (2023).

![图片](https://github.com/user-attachments/assets/d6ee8fb9-fdeb-46d8-a2b1-348aed0cbb03)

KexyBiscuit commented:
Smoke testing passed on Latitude D630.

![image_2024-10-10_05-40-12](https://github.com/user-attachments/assets/5991b621-57db-4f75-9ccd-525d91a55f32)

KexyBiscuit commented:
Smoke testing passed on ProArt B550-CREATOR.

![图片](https://github.com/user-attachments/assets/f9ced9f9-249f-4295-a3fb-a000b75ff762)

MingcongBai commented:
LGTM on LoongArch (THTF Chaorui L860-T2), also verified that PixArt patch v2 worked well with both the touchpad and a USB Logitech mouse.

eatradish committed:
linux-kernel: unset `CONFIG_USB_ONBOARD_DEV` to fix boot livekit on MacBook Pro Mid 2010

KexyBiscuit committed:
linux-kernel: update to 6.10.14

- Nearly identical with 6.10.14-rc1.
- Track patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.10.14, current HEAD is c588e0d40f688b9df73515b354a111b81240977e.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: update to 6.10.14

- Synchronize patches from linux-kernel.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit commented:
Smoke testing passed on AMD64 VMware VM.

- Product: VMware® Workstation 17 Pro
- Version: 17.6.1 build-24319023
- Host OS version: Windows 11 Enterprise, 64-bit (Build 27723.1000) 10.0.27723
- Accelerate 3D graphics: on
- Firmware type: UEFI
- Host VBS Mode: on

![image](https://github.com/user-attachments/assets/9cd8820c-0d81-416c-8a3c-7626994bef8a)

MingcongBai reviewed (approved):
Tested good with fix for MacBook Pro 15'' (Mid-2010).
